packages/browser-wallet/src/background/confirmation.ts

```diff
@@ -15,129 +16,134 @@ const isPendingCred = (cred: WalletCredential): cred is PendingWalletCredential
     cred.status === CreationStatus.Pending;
 const isPendingIdentity = (identity: Identity): identity is PendingIdentity =>
     identity.status === CreationStatus.Pending;
-const updateInterval = 10000;
-
-async function monitorCredentialStatus(
-    jsonRpcUrl: string,
-    { deploymentHash, ...info }: PendingWalletCredential,
-    genesisHash: string
-) {
-    const client = new JsonRpcClient(new HttpProvider(jsonRpcUrl, fetch));
-    async function loop() {
-        const network = await storedCurrentNetwork.get();
-        // Stop if the network has changed
-        if (!network || network.genesisHash !== genesisHash) {
-            return;
-        }
-        let repeat = true;
-        try {
-            const response = await client.getTransactionStatus(deploymentHash);
-            if (response?.status === TransactionStatusEnum.Finalized) {
-                const isSuccessful = Object.values(response?.outcomes || {}).some(
-                    (outcome) => outcome.result.outcome === 'success'
-                );
-                await updateCredentials(
-                    [
-                        {
-                            ...info,
-                            status: isSuccessful ? CreationStatus.Confirmed : CreationStatus.Rejected,
-                        },
-                    ],
-                    genesisHash
-                );
-                repeat = false;
-            }
-        } finally {
-            if (repeat) {
-                setTimeout(loop, updateInterval);
-            }
-        }
+const updateIntervalSeconds = 10;
+
+const ID_ALARM = 'check_id';
+const CRED_ALARM = 'check_cred';
+
+type ShouldRepeat = boolean;
+type CheckFun = () => Promise<ShouldRepeat>;
+
+const checkStatus = (checkFun: CheckFun, alarmId: string) => async () => {
+    const repeat = await checkFun();
+
+    if (!repeat) {
+        chrome.alarms.clear(alarmId);
```

Comment:
If I make multiple identities, does this trigger multiple confirmations for every pending identity?
+ Couldn't there be a "race condition" where this clears the alarm, but between this checking for pending and getting here, I have created another identity, which confirmation is then cleared by this?

Reply:
It seems to replace the alarm, and not duplicate it.

Reply:
Yes, that's also what the documentation states:

https://developer.chrome.com/docs/extensions/reference/alarms/#method-create

Reply:
Could the race condition not still occur? :thinking:

Reply:
I suppose it does. I'll see if I can come up with a solution. Can probably use the alarm name to hold some sort of state for both identities and credentials.

Reply:
Ideally I wouldn't have rewritten the confirmation loops to not be per identity/credential, and then you could do encode the identifier into the name, yes.
But with this I guess you could just not have a repeating alarm, but instead here schedule the next triggering, if there is still pending, That would remove that race condition, and worse case would be this delaying the next check.

Reply:
> Ideally I wouldn't have rewritten the confirmation loops to not be per identity/credential, and then you could encode the identifier into the name, yes.

My brain hurts trying to understand the double negative...

I have to rethink the solution anyway and not use alarms, due to their unreliability for these kinds of jobs. Maybe we could go for a solution that periodically checks while the popup is open? In practice, I don't think it would make much of a difference anyway?